libs/designer/src/lib/ui/settings/settingsection.tsx

```diff
@@ -332,6 +333,7 @@ const Setting = ({
     };
 
     const removeParamCallback = () => {
+      dispatch(storeStateToUndoRedoHistory({ type: updateParameterConditionalVisibility.type }));
```

Comment:
Difference in behaviour in that case:
1. If user clicks "show all" or "remove all", it'll save a state for each optional parameter, so it'll undo/redo parameters visibility one at a time instead of state before and after button clicks.
2. When user selects multiple options in the dropdown for optional parameters, it'll store state for each one and undo/redo them individually instead of multiple params together

Would this be preferable instead?